packages/spindle-ui/src/Modal/SemiModal.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { mergeRefs } from 'react-merge-refs';
+import { CrossBold } from '../Icon';
+import { IconButton } from '../IconButton';
+import { SubtleButton } from '../SubtleButton';
+
+type Size = 'large' | 'medium' | 'small';
+type Type = 'sheet' | 'popup';
+
+interface SemiModalProps extends React.DialogHTMLAttributes<HTMLElement> {
+  children?: React.ReactNode;
+  type?: Type;
+  size?: Size;
+  onClose?: (event: React.BaseSyntheticEvent) => void;
+}
+
+export interface DialogHTMLElement extends HTMLElement {
+  close?: () => void;
+  showModal?: (returnValue?: string) => void;
+  open?: boolean;
+}
+
+const BLOCK_NAME = 'spui-SemiModal';
+
+const Frame = forwardRef<DialogHTMLElement, SemiModalProps>(function SemiModal(
+  {
+    children,
+    className,
+    open,
+    size = 'large',
+    type = 'popup',
+    onClose,
+    ...rest
+  },
+  ref,
+) {
+  const [closing, setClosing] = useState(false);
+  const dialogEl = useRef<DialogHTMLElement>(null);
+
+  const FADE_OUT_ANIMATION = () => {
+    if (type === 'sheet') return 'spui-SemiModal-slide-out';
+    else return 'spui-SemiModal-fade-out';
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault(); // To be closed with the open prop
+    onClose && onClose(event);
+  };
+
+  const handleDialogClick = (event: React.MouseEvent<DialogHTMLElement>) => {
+    // Detect backdrop click
+    if (event.target === dialogEl.current) {
+      onClose && onClose(event);
+    }
+  };
```

Comment:
```suggestion
  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault(); // To be closed with the open prop
    onClose?.(event);
  };

  const handleDialogClick = (event: React.MouseEvent<DialogHTMLElement>) => {
    // Detect backdrop click
    if (event.target === dialogEl.current) {
      onClose?.(event);
    }
  };
```
IMO. optional chainingで省略しても良さそうです

Reply:
こちら修正しました！AppealModalの方は省略していなさそうだったので後ほど修正した方が良いでしょうか、、？

Reply:
ありがとうございます！
MUSTではないですが、お手すきで対応いただければ嬉しいです！🙏:hosupi: